app/controllers/application_controller.rb

```diff
@@ -1,3 +1,9 @@
 class ApplicationController < ActionController::Base
   before_action :authenticate_user!
-end
\ No newline at end of file
+
+  def execute_read_task
+    system('rake read')
+    redirect_to root_path
```

Comment:
How about using the redirect back option: https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back_or_to

Reply:
I am not quite sure which controller is best for this action, but we don't have much information at this stage - so this should be fine.